hibernate-core/src/main/java/org/hibernate/type/descriptor/jdbc/spi/JsonGeneratingVisitor.java

```diff
@@ -0,0 +1,252 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright Red Hat Inc. and Hibernate Authors
+ */
+package org.hibernate.type.descriptor.jdbc.spi;
+
+import org.hibernate.metamodel.mapping.CompositeIdentifierMapping;
+import org.hibernate.metamodel.mapping.EmbeddableMappingType;
+import org.hibernate.metamodel.mapping.EntityIdentifierMapping;
+import org.hibernate.metamodel.mapping.EntityMappingType;
+import org.hibernate.metamodel.mapping.ManagedMappingType;
+import org.hibernate.metamodel.mapping.MappingType;
+import org.hibernate.metamodel.mapping.SelectableMapping;
+import org.hibernate.metamodel.mapping.ValuedModelPart;
+import org.hibernate.metamodel.mapping.internal.EmbeddedAttributeMapping;
+import org.hibernate.metamodel.mapping.internal.SingleAttributeIdentifierMapping;
+import org.hibernate.type.BasicType;
+import org.hibernate.type.SqlTypes;
+import org.hibernate.type.descriptor.WrapperOptions;
+import org.hibernate.type.descriptor.java.BasicPluralJavaType;
+import org.hibernate.type.descriptor.java.JavaType;
+import org.hibernate.type.descriptor.jdbc.AggregateJdbcType;
+import org.hibernate.type.descriptor.jdbc.ArrayJdbcType;
+import org.hibernate.type.descriptor.jdbc.JdbcType;
+import org.hibernate.type.descriptor.jdbc.JsonJdbcType;
+import org.hibernate.type.format.JsonDocumentWriter;
+
+import java.io.IOException;
+import java.lang.reflect.Array;
+
+import static org.hibernate.type.descriptor.jdbc.StructHelper.getSubPart;
+
+/**
+ * Stateless helper class to serialize managed type values to JSON.
+ */
+public class JsonGeneratingVisitor {
+
```

Comment:
```suggestion

	public static final JsonGeneratingVisitor INSTANCE = new JsonGeneratingVisitor();
	
	protected JsonGeneratingVisitor() {}
```

Reply:
Mmh I have instantiated the visitor in the classes that need it - why would we always initialize a static instance instead?

Reply:
To avoid creating multiple instances of this stateless class :)